File: SQLDisplayControl/Word.vb
Option Strict Off
Option Explicit On
Namespace DBDiff
    Friend Class Word

        Public Enum WordClassType
            SQLWord = 1
            FunctionWord = 2
            StringWord = 3
            NumberWord = 4
            CommentWord = 5
            OperatorWord = 6
            SysTablesWord = 7
            TypeWord = 8
        End Enum

        Private oWordText As String
        Private oWordType As WordClassType
        Private oLen As Integer
        Public ReadOnly Property Size() As Integer
            Get
                Size = oLen
            End Get
        End Property


        Public Property Text() As String
            Get
                Text = oWordText
            End Get
            Set(ByVal Value As String)
                oWordText = Value
                oLen = Len(oWordText)
            End Set
        End Property


        Public Property WordType() As WordClassType
            Get
                WordType = oWordType
            End Get
            Set(ByVal Value As WordClassType)
                oWordType = Value
            End Set
        End Property

        Public ReadOnly Property IsOpenWordType() As Boolean
            Get
                IsOpenWordType = (oWordType = WordClassType.StringWord) Or (oWordType = WordClassType.CommentWord)
            End Get
        End Property
    End Class
End Namespace

File: SQLRichControl/Word.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace SQLRichControl
{
    internal class Word:IComparable<Word>
    {
        public enum WordClassType
        {
            SQLWord = 1,
            FunctionWord = 2,
            StringWord = 3,
            NumberWord = 4,
            CommentWord = 5,
            OperatorWord = 6,
            SysTablesWord = 7,
            TypeWord = 8,
            SysProcWord = 9
        }

        private string text;
        private WordClassType type;
        private int len;

        public string Text
        {
            get { return text; }
            set 
            { 
                text = value;
                len = text.Length;
            }
        }

        public WordClassType Type
        {
            get { return type; }
            set { type = value; }
        }

        public int Size
        {
            get { return len; }
        }
        
        public bool IsOpenWordType
        {
            get
            {
                return (type == WordClassType.StringWord) || (type == WordClassType.CommentWord);
            }
        }

        public int CompareTo(Word other)
        {
            return text.CompareTo(other.text);
        }

    }
}
